app/src/main/java/com/github/orkest/Model/FireStoreDatabaseAPI.kt

```diff
@@ -131,29 +142,97 @@ class FireStoreDatabaseAPI {
         return db.collection(path).document(username)
     }
 
+    //===========================SONG POST OPERATIONS======================
 
+    private fun getPostCollectionRef(username: String): CollectionReference{
+        val firstLetter = username[0].uppercase()
+        //TODO: Discuss other option: "posts/user-$firstLetter/$username"
+        //Chose this for now because easier for group queries
+        // But question: what happens when we get the docRef of one user
+        val path = "user/user-$firstLetter/users/$username/posts"
+        return db.collection(path)
+    }
 
 
+    /**
+     * @param post the post we want to add in the database
+     * @return a completable future that completes to true only if post is correctly added
+     */
+    fun addPostInDataBase(post: Post): CompletableFuture<Boolean> {
+        val future = CompletableFuture<Boolean>()
+        val postDocument = getPostCollectionRef(post.username).document(post.date.toString())
 
+        postDocument.set(post).addOnSuccessListener { future.complete(true) }
+            .addOnFailureListener { future.completeExceptionally(it) }
 
+        return future
+    }
 
+    /**
+     * @param post the post we want to delete in the database
+     * @return a completable future that completes to true only if post is correctly deleted
+     */
+    fun deletePostInDataBase(post: Post): CompletableFuture<Boolean> {
+        val future = CompletableFuture<Boolean>()
+        val postDocument = getPostCollectionRef(post.username).document(post.date.toString())
 
+        postDocument.delete().addOnSuccessListener { future.complete(true) }
+            .addOnFailureListener { future.completeExceptionally(it) }
 
+        return future
+    }
 
+    /**
+     * @param username the username of the user we want to get the posts from
```

Comment:
very small detail, but add a colon after the name of the param.
@param username: the username